src/config.ts

```diff
@@ -64,11 +64,13 @@ export interface Config extends AuthenticationConfig {
 
 // Default values for configuration for a profiler.
 export const defaultConfig: Config = {
-  projectId: '{{projectId}}',
   logLevel: 1,
+  projectId: '{{projectId}}',
```

Comment:
The projectId is a part of the url to communicate with the profiler API (for example, https://cloudprofiler.googleapis.com/v2/projects/{{projectId}}/profiles),  so it needs to be set.
Can this still be omitted? Can we trust this to be filled in correctly?

Reply:
That's the case for [trace as well](https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects/patchTraces). This particular string is an implementation detail of the @ google-cloud/common repo, and is likely to be changed going forward. I would like to avoid promulgating it elsewhere. Please avoid if at all possible.

Reply:
Perhaps the `projectIdRequired` property on `ServiceAuthenticationConfig` is a way to avoid specifying this. Can you take a look?

Reply:
That should work.